proposals/pr-contrib-guide.md

```diff
@@ -0,0 +1,315 @@
+
+## Specification
+
+This section defines:
+
+* [Current process](#current-process)
+* [Proposed improvements](#proposed-improvements)
+
+### Current Process
+
+Customer feature requests are reviewed for general suitability and uniqueness,
+but are not actively commented on by maintainers to move the feature forward to
+an implementation. Pull requests are reviewed in an *ad hoc* fashion by various
+members of the maintainer team, but similarly are not actively processed to
+continue towards acceptance. This has contributed to the current status of 189
+pull requests open in the AWS CLI GitHub repository.
+
+### Proposed Improvements
+ 
+Maintainers that actively engage with the community contributors to identify
+suitable implementations for requested features and provide gudiance during the
+process will result in a higher percentage of successful community
+contributions. In addition, we will publish a contributing guide that describes
+the life cycle of a feature request that can culminate in a code contribution
+from the community through a GitHub pull request. The guide describes the triage
+and review of open feature requests with the goal of identifying issues where
+community contributions would be welcome. It defines what is required to move a
+feature request to be available for community contribution. It documents the
+review process and sets community expectations around communications to move a
+contribution forward. Figure A demonstrates a high level view of the proposed
+process.
+
+The maintainers will actively review GitHub issues for feature requests to
+determine if they are available for contribution. The criteria used to make that
+determination include:
+
+1. The change will be made on part of the code base that the maintainers have
+   determined to be available. See the FAQ for more details.
+1. There is evidence that the change will be impactful for a significant number
+   of users beyond the individual that proposed it.
+1. The background, description, and use cases for the feature are clearly
+   defined.
+1. There are test cases and edge cases defined.
+1. There is consensus with the maintainers on an implementation plan.
+
+Once these criteria have been met, an issue is available for the community to
+contribute. If it's not something that the maintainers feel should be added, the
+issue should be left for community discussion and upvotes. A comment should be
+added by the maintainers to describe how users can advocate for the feature with
+more use cases, or vote for the issue using reactions. The issue is subject to
+existing stale and ancient issue closure mechanisms.
+
+When an issue is being worked on by the community, maintainers will provide
+guidance and support during the contribution process. They will engage with the
+community during discussion of implementation details or issues that arise
+during the contribution. Notably, we will initially respond within [TBD]
```

Comment:
Any time period here is going to need to consider how often we will be looking at these. Ad-hoc is not going to work I don't think.

Reply:
Agree, discussed. Resolving.